src/Java.Interop/Java.Interop/ManagedPeer.cs

```diff
@@ -19,18 +20,27 @@ namespace Java.Interop {
 
 		static  readonly    JniPeerMembers  _members        = new JniPeerMembers (JniTypeName, typeof (ManagedPeer));
 
-		static ManagedPeer ()
+		static unsafe ManagedPeer ()
```

Comment:
Alternatively, instead of introducing a "blittable method registration struct", we can instead use `[UnmanagedFunctionPointer]` to coerce NativeAOT to produce the required stubs:

```diff
diff --git a/src/Java.Interop/Java.Interop/ManagedPeer.cs b/src/Java.Interop/Java.Interop/ManagedPeer.cs
index eb43a9e8..044b9424 100644
--- a/src/Java.Interop/Java.Interop/ManagedPeer.cs
+++ b/src/Java.Interop/Java.Interop/ManagedPeer.cs
@@ -6,6 +6,7 @@ using System.Linq;
 using System.Linq.Expressions;
 using System.Reflection;
 using System.Reflection.Emit;
+using System.Runtime.InteropServices;
 using System.Runtime.Serialization;
 using System.Text;
 
@@ -50,6 +51,7 @@ namespace Java.Interop {
 		const string ConstructSignature = "(Ljava/lang/Object;Ljava/lang/String;Ljava/lang/String;[Ljava/lang/Object;)V";
 
 		// TODO: Keep in sync with the code generated by ExportedMemberBuilder
+		[UnmanagedFunctionPointer (CallingConvention.Winapi)]
 		delegate void ConstructMarshalMethod (IntPtr jnienv,
 				IntPtr klass,
 				IntPtr n_self,
@@ -177,6 +179,7 @@ namespace Java.Interop {
 
 		const   string  RegisterNativeMembersSignature  = "(Ljava/lang/Class;Ljava/lang/String;Ljava/lang/String;)V";
 
+		[UnmanagedFunctionPointer (CallingConvention.Winapi)]
 		delegate void RegisterMarshalMethod (IntPtr jnienv,
 				IntPtr klass,
 				IntPtr n_nativeClass,
```

Reply:
I'm uncertain how much I want to push unmanaged function pointers everywhere.  I like the idea, but there are "unknown" implications, not least of which is `JniNativeMethodRegistrationArguments`.

Reply:
Turns out that for NativeAOT, `JniNativeMethodRegistrationArguments` is unusable, because to *get* there we need to pass through `Type.GetType("TypeToRegister, DeclaringAssembly")`, which throws `System.IO.FileNotFoundException: Could not resolve assembly 'DeclaringAssembly'.`.

Suggesting that there isn't actually any point in a blittable equivalent to `JniNativeMethodRegistrationArguments`, because we're gonna need to use `Java_…` native methods *anyway*.  It's not optional for NativeAOT, it's *required*.

Reply:
…and I in turn replaced the "new" (now removed) `JniRuntime.JniTypeManager.RegisterNativeMembers()` overload with an alternate approach involving `JniRuntime.JniTypeManager.GetTypeFromAssemblyQualifiedName()`, and "bulking up" the typemap dictionary within `NativeAotTypeManager`.  This in no way verifies or removes the need for a blittable registration struct; it just further punts it down the line.

That said, a blittable struct would presumably be more efficient…